File: Dias/016.js
/* Exercicio feito em Jscript 
https://www.hackerrank.com/challenges/migratory-birds/problem
*/

function migratoryBirds(arr) {
    let passarosCont = {};

    for (let i = 1; i <= 5; i++) {
        passarosCont[i] = 0;
    }

    for (let passaro of arr) {
        passarosCont[passaro]++;
    }

    let maisPassaro = 1;
    for (let i = 2; i <= 5; i++) {
        if (passarosCont[i] > passarosCont[maisPassaro]) {
            maisPassaro = i;
        }
    }

    return maisPassaro;
}

// Teste de usos para verificar se esta saindo corretamente
const arr = [1, 4, 4, 4, 5, 3];
console.log(migratoryBirds(arr)); 

File: Dias/016.py
'''Exercicio feito em Python 
https://www.hackerrank.com/challenges/migratory-birds/problem

'''


def migratoryBirds(arr):
    passaros_Cont = {1: 0, 2: 0, 3: 0, 4: 0, 5: 0}

    for passaro in arr:
        passaros_Cont[passaro] += 1

    maisPassaro = 1
    for i in range(2, 6):
        if passaros_Cont[i] > passaros_Cont[maisPassaro]:
            maisPassaro = i

    return maisPassaro

# Exemplo de uso:
arr = [3, 5, 2, 1, 3, 4]
print(migratoryBirds(arr))  
